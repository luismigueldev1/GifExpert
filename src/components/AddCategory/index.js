import React, { useState } from "react";
import PropTypes from "prop-types";

function AddCategory({ setCategories }) {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (event) => {
    setInputValue(event.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    const fixedInputValue = inputValue.trim();

    if (fixedInputValue) {
      setCategories((categories) => {
        const isAlreadyExist = categories.find(
          (cat) => cat === fixedInputValue
        );
        if (isAlreadyExist) return [...categories];
        return [fixedInputValue, ...categories];
      });
      setInputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="add-category">Search gifs: </label>
      <input
        type="text"
        name="add-category"
        placeholder="Mario bros"
        value={inputValue}
        onChange={handleInputChange}
      />
      <button type="submit">Search</button>
    </form>
  );
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
