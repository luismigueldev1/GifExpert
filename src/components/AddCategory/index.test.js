import { shallow } from "enzyme";
import AddCategory from ".";

describe("Pruebas en AddCategory", () => {
  const setCategories = jest.fn();

  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test("debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de cambiar la caja de texto ", () => {
    const input = wrapper.find("input");
    const value = "Hola Mundo";
    input.simulate("change", { target: { value: value } });
  });

  test("NO debe de postear la informaciÃ³n on submit si no se le envia un valor en el input", () => {
    wrapper.find("form").simulate("submit", {
      preventDefault() {},
    });

    expect(setCategories).not.toHaveBeenCalled();
  });

  test("Debe de llamar el setCategories y limpiar la caja de texto", () => {
    const value = "Hola Mundo";
    const input = wrapper.find("input");
    input.simulate("change", { target: { value: value } });

    wrapper.find("form").simulate("submit", {
      preventDefault() {},
    });
    //expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    //expect(input.text()).toBe("");
    expect(input.prop("value")).toBe("");
  });
});
