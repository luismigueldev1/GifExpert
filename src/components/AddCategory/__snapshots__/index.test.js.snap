// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AddCategory debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="add-category"
  >
    Search gifs: 
  </label>
  <input
    name="add-category"
    onChange={[Function]}
    placeholder="Mario bros"
    type="text"
    value=""
  />
  <button
    type="submit"
  >
    Search
  </button>
</form>
`;
